subprojects/platform-jvm/src/main/java/org/gradle/jvm/tasks/Jar.java

```diff
@@ -202,4 +218,25 @@ public CopySpec getMetaInf() {
     public CopySpec metaInf(Closure<?> configureClosure) {
         return ConfigureUtil.configure(configureClosure, getMetaInf());
     }
+
+    /**
+     * Provides a comparator that sorts the contents of a Jar/Ear/War so that META-INF is always first.
+     */
+    private static class JarContentsComparator implements Comparator<RelativePath> {
+        @Override
+        public int compare(RelativePath o1, RelativePath o2) {
+            boolean o1MetaInf = isMetaInf(o1.getSegments()[0]);
```

Comment:
For reference, here is the specification for jar files: http://docs.oracle.com/javase/7/docs/technotes/guides/jar/jar.html
Note that the file `INDEX.LIST` would come before `MANIFEST.MF` when we just sort with the natural order.